src/driver.ts

```diff
@@ -39,44 +41,107 @@ export interface Driver {
   unsupportedFeatures?: {
     security?: UnsupportedFeatureBehaviour;
   };
+  forceGenerateTypes?: boolean; // output the type file even if it would have been already generated
 }
 
 function emitAllStatusCodes() {
   return true;
 }
 
-function resolveModule(fromModule: string, toModule: string): string {
-  if (!toModule.startsWith('.')) {
-    return toModule;
-  }
+function defaultResolve() {
+  return undefined;
+}
 
-  const p = path.relative(path.dirname(fromModule), toModule);
-  if (p[0] === '.') {
-    return p;
+export function localResolve(ref: string) {
```

Comment:
I'm guessing  other than `local` the only remaining option to do a valid `$ref` would be the URL kind... if that's the case, why expose all this? Couldn't we just figure out what  type  the  `$ref` is and then  resolve it automatically?

Reply:
not sure I follow. As a guess: you are asking why we expose `resolve` at all and not just do the right thing automatically?

as a default that would be nice -- however I did not want to change the behaviour of the resolution for existing users. 

In the greater scheme of things `resolve` allows also, in addition to fetching files from urls and more importantly defining *how* and where that happens, mapping names to custom implementations or shared implementations. 

For example we can do https://github.com/smartlyio/facebook-blueprint/pull/70/files#diff-ea1fc938c4289be40df8874165cbd861R3 which we cannot just "do-the-right-things"

Reply:
your guess it correct

> defining how and where that happens, mapping names to custom implementations or shared implementations

makes sense 👍